BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -47,29 +48,39 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
+        const string pattern = @"^did\:web\:(.+?)\:dids\:(.{3})(.+)$";
+        var matches = Regex.Matches(stringValue, pattern, RegexOptions.IgnoreCase);
 
-        var concatenation = Base58.Bitcoin.Decode(stringValue.AsSpan(3)).ToArray();
-        var hashedPublicKey = concatenation[..20];
-        var givenChecksum = concatenation[20..];
+        try
+        {
+            var realm = matches.First().Groups[2].Value;
+            var concatenation = Base58.Bitcoin.Decode(matches.First().Groups[3].Value).ToArray();
 
-        var realmBytes = Encoding.UTF8.GetBytes(realm);
-        var correctChecksum = CalculateChecksum(realmBytes, hashedPublicKey);
+            var hashedPublicKey = concatenation[..20];
+            var givenChecksum = concatenation[20..];
+
+            var realmBytes = Encoding.UTF8.GetBytes(realm);
+            var correctChecksum = CalculateChecksum(realmBytes, hashedPublicKey);
 
-        var checksumIsValid = givenChecksum.SequenceEqual(correctChecksum);
+            var checksumIsValid = givenChecksum.SequenceEqual(correctChecksum);
 
-        return lengthIsValid && checksumIsValid;
+            return lengthIsValid && checksumIsValid;
+        }
+        catch (IndexOutOfRangeException _)
```

Comment:
1. Remove the `_` to get rid of the compiler warning
2. Instead of provoking the exception, we should do length checks beforehand. In general you should try to avoid exceptions whenever possible

Reply:
We use this function: https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.matches?view=net-8.0#system-text-regularexpressions-regex-matches(system-string-system-string-system-text-regularexpressions-regexoptions-system-timespan)

There does not seem to be a way to avoid the exceptions when the input does not match the expected format. The type I put in the catch block in wrong, granted, but I don't see a simple way to avoid it.

Reply:
There are 3 exceptions documented:
1. `ArgumentException`: This one is thrown when the given regular expression is invalid. But if the regular expression actually IS invalid, then we will notice this pretty quickly, as this logic is used in pretty much every test.
2. `ArgumentNullException`: This one is thrown when either `input` or `pattern` is `null`. We can simply check `input` for `null` before we check the regex. And for `pattern`, the same arguments as in (1) apply.
3. `ArgumentOutOfRangeException`: Here the same arguments as in (1) apply.